py/bodylineleft.py

```diff
@@ -1,19 +1,90 @@
+                                with tr().add(td(cls="td-valign detail-pad")):
+                                    #Iterates through the details array and generates HTML depending on object type.
+                                    for line in entry.details:
+                                        if isinstance(line, Link):
+                                            with p(cls="details").add(u()):
+                                                a(line.text, cls="link", href=line.url, target="_blank")
+                                        elif isinstance(line, Linebreak):
+                                            for i in range(line.numBreaks):
+                                                br()
+                                        elif isinstance(line, Text):
+                                            p(line.text, cls="details")
+                                        else:
+                                            raise TypeError("Elements of details must be either a Text, Linebreak or Link object.")
+                            #Iterates through the body of the entry to create HTML for rendering a block of content or an image.
+                            for elem in entry.body:
+                                if isinstance(elem, Content):
+                                    with tr().add(td(cls="td-valign left-desc-pad")):
+                                        for line in elem.desc:
+                                            if isinstance(line, Link):
+                                                with p(cls="description").add(u()):
+                                                    a(line.text, cls="link", href=line.url, target="_blank")
+                                            elif isinstance(line, Linebreak):
+                                                for i in range(line.numBreaks):
+                                                    br()
+                                            elif isinstance(line, Text):
+                                                p(line.text, cls="description")
+                                            else:
+                                                raise TypeError("Elements of Content object must be either a Text, Link or Linebreak object.")
+                                elif isinstance(elem, Image):
+                                    with tr().add(td(cls="td-valign entry")):
+                                        img(cls="img-left", src=elem.url, alt=elem.alt)
+                                else:
+                                    raise TypeError("Elements of body must be either Content or Image objects")
+                    #If the entry is an Image, then the image is rendered or the alt is displayed.
+                    elif isinstance(entry, Image):
+                        with tr().add(td(cls=entry_class)):
+                            img(cls="img-left", src=entry.url, alt=entry.alt)
+                    else:
```

Comment:
Will there ever be a situation where the entries are neither Entry or Image?

Reply:
No

Reply:
If there isn't is this else clause necessary?

Reply:
It is there to raise errors when the correct types aren't present rather than generating incorrect HTML.